pkg/reconciler/v1alpha1/build/build.go

```diff
@@ -86,25 +88,40 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 		return nil
 	}
 
-	pod, err := c.reconcileBuildPod(build)
-	if err != nil {
-		return err
-	}
+	if build.Rebasable() {
+		//todo how to handle errors
 
-	if build.MetadataReady(pod) {
-		image, err := c.MetadataRetriever.GetBuiltImage(build)
+		image, err := c.ImageRebaser.Rebase(build)
 		if err != nil {
 			return err
```

Comment:
Maybe use a NewPermanentError to prevent it from being re-enqued?

Reply:
🙏